src/Language/PureScript/TypeChecker/TypeSearch.hs

```diff
@@ -62,12 +66,21 @@ checkSubsume env nextVar userT envT = checkInEnvironment env nextVar $ do
   elab <- subsumes envT' userT'
   subst <- gets TC.checkSubstitution
   let expP = P.overTypes (P.substituteType subst) (elab dummyExpression)
-  Entailment.replaceTypeClassDictionaries False expP
+
+  -- Now check that any unsolved constraints have not become impossible
+  (traverse_ . traverse_) (\(_, context, constraint) -> do
+    let constraint' = P.mapConstraintArgs (map (P.substituteType subst)) constraint
+    flip evalStateT Map.empty . evalWriterT $
+      Entailment.entails True False (P.TypeClassDictionary constraint' context [])) cons
```

Comment:
Maybe add a comment here, or haddocks on the `entails` definition for the two Booleans? The first one is `shouldGeneralize` and the second one is `defer`.